Adding python3.9 to travis-ci config

nomadmtb opened the pull request:
Since python3.9 was released on Oct 5th 2020, it might be good to start running the test suite with that new version now.

AaronAtDuo commented:
Apparently Python 3.9 support on Travis should be ready within the next day or so:
https://travis-ci.community/t/python-3-9-0-build/10091/11

nomadmtb commented:
Thanks for sharing that @AaronAtDuo!  I'll keep my eyes on that thread and re-run it once 3.9 gets added.

nomadmtb commented:
Okay this should be ready for review again.  It looks like 3.9 is now enforcing valid hostnames because of a CVE posted https://github.com/python/cpython/commit/9165addc22d05e776a54319a8531ebd0b2fe01ef which explains why some of our tests were failing initially when I added 3.9.  I updated this PR to include a correction on a few unittests where we were instantiating an HTTPConnection class with a bad hostname.
